### ISSUE TITLE
New Boss Idea

### ISSUE THREAD

**Issue by profilename14** [Tslat/Advent-Of-Ascension#2784]
I had an interesting dream today in which I got the idea for an entire 6 phase boss fight that seemed pretty well designed. Basically, the boss took the form of 3 large flames, one orange, one blue, and one green. The first three phases involve fighting the flames individually, while phases 4-6 involve fighting all at once while picking them off in any order. Each flame had the same HP, so I'd assume each one has 500 Health and the boss has 3000 total. The fight took place in a large, walled, circular arena, where the boss stays in the center for the first three phases.

The first phase has you fight the orange flame which shoots spreads of 3 sword projectiles at you, aiming where you are going to be. After firing 5 shots, there is a short cooldown in which you are free to attack. To dodge these projectiles, the player must try to shift to the side until the flame fires, reverse direction until they fire again, and so on at the right speed to throw off the aim of the boss. The player can use ranged weapons during attacks and can switch to melee when the boss is vulnerable.

The second phase has you fight the blue flame, who shoots a constant blue flame thrower to the edge of the arena. It deals armor-piercing damage and, at moderate speed (slower than the player) turns around clockwise or counterclockwise, whichever reaches the player faster. To avoid the attack, the player must go closer to the flame and get to the other side of the flame thrower, where they have a window to continue attacking before they have to sprint away again. The boss will occasionally stop using its flame thrower for a moment if the player is too far ahead and try to aim it at the player, so they must be ready to dodge this as well.

The third phase involves the green flame shooting beams of energy in 4 cardinal directions, turning at a relatively fast speed. However, unlike the last two attacks, these beams can be jumped over. The player must attack while making sure t jump at the right time to dodge each beam. In addition, with a sound cue, the boss will occasionally send out a circular laser ring that the player must try to jump over while also evading the four carinal beams. The player must try to attack with melee or ranged weapons as they are evading attacks.

Finally, in the fourth phase, each flame materializes in a triangle, pointing a stream of swords, flamethrower, and beam at another to make the walls. In this phase, the boss moves around the arena and tries to catch the player in its walls to damage them. Each wall has a weakness that allows the player to avoid them. The stream of swords sent by the orange flame has windows in which no swords are being fired, so the player can sprint through at the right time. the player can also block the swords, unlike the other two attacks. The blue wall lags behind the boss, leading to it being possible for the player to run through the sides of the wall to escape when the boss is moving. Finally, the green wall can be jumped over, but there is an interval in which it cannot because it intensifies and catches a jumping player. One risky strategy a player can use to get through this phase quickly is by getting inside the triangle and staying in the center while using melee attacks.

Phase 5 starts when one of the flames is defeated. The flame that was previously emitting its attack against the defeated flame becomes the "dominant" boss of this phase, while the one that was previously having an attack being emitted at it by the defeated flame becomes the "supportive" boss of this phase. 

The dominant flame generally moves around and aims at the player. A dominant green flame fires a laser that can be jumped over, moving from end to end of an arc-shaped area (basically, say firing at say 60 degrees to 70 degrees to 60 degrees if the player is moving still) shifting the beam to fire around the player if they move around. A dominant orange flame, like the first phase, fires bursts of swords the player must strafe to dodge. A dominant blue flame fires occasional blasts of blue fire similar to its special attack in phase two, although the flames can also move toward the player for a short time before fading.

The supportive flame generally moves in a circle around the dominant flame while making a wall the player must evade. Orange and Green Supportive flames generally work the same as their wall versions at the start of phase four, while a blue supportive flame works sort of like it does in phase two where it is slow-moving.

Finally, Phase 6 starts when either the supportive or dominant flame is defeated. This plays out like phases 1, 2, or 3, depending on what flame remains, but with less forgiving attacks that cause higher damage. The orange flame has better aim, the blue flame approaches the player faster and uses its special attack more often, and the green flame uses its special attack more often while sometimes intensifying its four beams to prevent them from being jumped over.

When the final flame is defeated, you get a weapon based on that boss. This means that you can pick and choose which weapon you obtain based on how you go through the last 3 phases. Defeating the orange flame last gives you a greatsword made out of the fire which, when used with a right-click, fires a crescent of flame at the cost of energy and extra durability. Defeating the blue flame last gives you a staff that fires a piercing gout of blue fire that harms an area of enemies. Finally, defeating the green flame last gives you a blaster that fires a piercing, long-range laser beam.

I know we have quite a few bosses in this mod already so it is perfectly fine if this suggestion is not accepted. I feel like this would be a cool idea for the four guardians if it were adapted to four bosses. If this boss is actually taken into the mod, I feel like it would best fit in Lelyetia, as that is the only single boss dimension left that a fire-themed boss could fit into. Maybe it could be fought on the underside or something.

**Comment by Cruuk** [comment]
Lelyetia isn't fire themed anymore since 3.2. Closest thing besides Nether is Barathos

**Comment by profilename14** [comment]
Yeah, I was considering Barathos due to the lava-filled underworld and desert-like environment, but the main problem is simply that there are already 2 bosses there. The Nether is even worse, as it has three bosses counting the vanilla Wither. However, a boss doesn't necessarily have to completely align with the theme of the dimension anyway, so perhaps it could go somewhere else. Looking through the dimensions, the top three picks would probably be something between Creeponia, Runandor, or Crystevia. Creeponia is supposed to be a "creeper" dimension but it also has a sort of explosive theme, which is at least somewhat close to the boss's theming. It might also fit in Runandor or Crystevia because of the boss's sort of magical theming, being made of three fire elementals. Crystevia has the bonus of having a variety of colors in its general theming, including red, blue, and green, which would let the boss fit visually as well. Of course, it would probably be better to somehow work these ideas into an existing boss, so this is just if it gets added normally.

**Comment by Cruuk** [comment]
Or make it special and have it require multiple dims
